tests/mfx_dispatch/linux/mfx_dispatch_test_cases_libs.cpp

```diff
@@ -0,0 +1,282 @@
+    EXPECT_CALL(mock, MFXQueryVersion).Times(AtLeast(1)).WillRepeatedly(DoAll(SetArgPointee<1>(nullversion), Return(MFX_ERR_NONE)));
+    EXPECT_CALL(mock, MFXClose).Times(AtLeast(1));
+    EXPECT_CALL(mock, dlclose).Times(AtLeast(1));
+
+    mfxStatus sts = MFXInit(impl, &ver, &session);
+    EXPECT_EQ(sts, MFX_ERR_UNSUPPORTED);
+}
+
+TEST_F(DispatcherLibsTest, ShouldFailIfImplNotSupportedByLib)
+{
+    MockCallObj& mock = *g_call_obj_ptr;
+
+    ver = {{MFX_VERSION_MINOR, MFX_VERSION_MAJOR}};
+    mock.emulated_api_version = ver;
+
+    mfxVersion nullversion = {{0}};
+
+    EXPECT_CALL(mock, dlopen).Times(AtLeast(1)).WillRepeatedly(Return(MOCK_DLOPEN_HANDLE));
+    EXPECT_CALL(mock, dlsym).Times(AtLeast(1)).WillRepeatedly(Invoke(&mock, &MockCallObj::EmulateAPI));
+    EXPECT_CALL(mock, MFXInitEx).Times(AtLeast(1)).WillRepeatedly(DoAll(SetArgPointee<1>(MOCK_SESSION_HANDLE), Return(MFX_ERR_NONE)));
+    EXPECT_CALL(mock, MFXQueryIMPL).Times(AtLeast(1)).WillRepeatedly(Return(MFX_ERR_UNSUPPORTED));
+    EXPECT_CALL(mock, MFXQueryVersion).Times(AtLeast(1)).WillRepeatedly(Invoke(&mock, &MockCallObj::ReturnEmulatedVersion));
+    EXPECT_CALL(mock, MFXClose).Times(AtLeast(1));
+    EXPECT_CALL(mock, dlclose).Times(AtLeast(1));
+
+    mfxStatus sts = MFXInit(impl, &ver, &session);
+    EXPECT_EQ(sts, MFX_ERR_UNSUPPORTED);
+}
+
+TEST_F(DispatcherLibsTest, ShouldSucceedIfFirstLibBadSecondLibGood)
+{
+    MockCallObj& mock = *g_call_obj_ptr;
+
+    ver = {{MFX_VERSION_MINOR, MFX_VERSION_MAJOR}};
+    mock.emulated_api_version = ver;
+
+    EXPECT_CALL(mock, dlopen).Times(AtLeast(1)).WillOnce(Return(nullptr)).WillRepeatedly(Return(MOCK_DLOPEN_HANDLE));
+    EXPECT_CALL(mock, dlsym).Times(AtLeast(1)).WillRepeatedly(Invoke(&mock, &MockCallObj::EmulateAPI));
+    EXPECT_CALL(mock, MFXInitEx).Times(AtLeast(1)).WillRepeatedly(DoAll(SetArgPointee<1>(MOCK_SESSION_HANDLE), Return(MFX_ERR_NONE)));
+    EXPECT_CALL(mock, MFXQueryIMPL).Times(AtLeast(1)).WillRepeatedly(Return(MFX_ERR_NONE));
+    EXPECT_CALL(mock, MFXQueryVersion).Times(AtLeast(1)).WillRepeatedly(Invoke(&mock, &MockCallObj::ReturnEmulatedVersion));
+
+    mfxStatus sts = MFXInit(impl, &ver, &session);
+    EXPECT_EQ(sts, MFX_ERR_NONE);
+
+    EXPECT_CALL(mock, dlopen).Times(AtLeast(1)).WillRepeatedly(Return(MOCK_DLOPEN_HANDLE));
+    EXPECT_CALL(mock, dlsym).Times(AtLeast(1)).WillOnce(Return(nullptr)).WillRepeatedly(Invoke(&mock, &MockCallObj::EmulateAPI));
+    EXPECT_CALL(mock, MFXInitEx).Times(AtLeast(1)).WillRepeatedly(DoAll(SetArgPointee<1>(MOCK_SESSION_HANDLE), Return(MFX_ERR_NONE)));
+    EXPECT_CALL(mock, MFXQueryIMPL).Times(AtLeast(1)).WillRepeatedly(Return(MFX_ERR_NONE));
+    EXPECT_CALL(mock, MFXQueryVersion).Times(AtLeast(1)).WillRepeatedly(Invoke(&mock, &MockCallObj::ReturnEmulatedVersion));
+    EXPECT_CALL(mock, dlclose).Times(AtLeast(1));
+
+    sts = MFXInit(impl, &ver, &session);
+    EXPECT_EQ(sts, MFX_ERR_NONE);
+}
+
+
```

Comment:
Other tests:
- Does MFXInitEx implemented in dispatcher pass mfxInitParam* untouched?
- Does dispatcher pass mfxSession* allocated inside the library to other library calls untouched?

Reply:
Should create an issue about this once the current PR is merged.